integration-tests/search.test.ts

```diff
@@ -513,3 +514,242 @@ describe('search', () => {
         });
     });
 });
+
+describe('searches with patient permission levels', () => {
+    let adminClient: AxiosInstance;
+    let encounterResource: any;
+
+    beforeAll(async () => {
+        adminClient = await getFhirClient('fhirUser user/*.*', true);
+        const encounterResourceJson = {
+            resourceType: 'Encounter',
+            status: 'in-progress',
+            class: {
+                system: 'http://terminology.hl7.org/CodeSystem/v3-ActCode',
+                code: 'IMP',
+                display: 'inpatient encounter',
+            },
+            subject: {
+                reference: `Patient/92e0d921-bb19-4cae-a3cc-9d3c5bcf7a39`,
+                display: 'Sherlock Holmes',
+            },
+        };
+        encounterResource = (await adminClient.post('Encounter', encounterResourceJson)).data;
+        await waitForResourceToBeSearchable(adminClient, encounterResource);
+    });
+
+    afterAll(async () => {
+        await adminClient.delete(`Encounter/${encounterResource.id}`);
+    });
```

Comment:
you can nest this at a higher level so that we only do this once across all tests
```
describe
  before..
  after...
  <existing describe tests>
```